Plugin/ClearQuote.php

```diff
@@ -25,14 +28,69 @@
  */
 class ClearQuote
 {
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var \Magento\Quote\Model\Quote|null
+     */
+    private $quote_to_restore;
+
+    /**
+     * @param CartHelper $cartHelper
+     */
+    public function __construct(
+        CartHelper $cartHelper
+    )
+    {
+        $this->cartHelper = $cartHelper;
+    }
+
+    /**
+     * @param CheckoutSession $subject
+     * @return CheckoutSession
+     * @throws \Exception
+     */
+    public function beforeClearQuote(CheckoutSession $subject)
+    {
+        // We don't want to clear quote
+        // in Product page checkout (PPC) flow
+        $this->quote_to_restore = null;
+        $current_quote_id = $subject->getQuote()->getId();
+        $order_quote_id = $subject->getLastSuccessQuoteId();
+        if (!$current_quote_id || !$order_quote_id && $current_quote_id == $order_quote_id) {
```

Comment:
Not sure I understand this condition. Can you explain?

Reply:
Uf, it's a type. Will fix.